src/XRP/serializer.ts

```diff
@@ -84,20 +67,23 @@ const serializer = {
     }
     object.Fee = this.xrpAmountToJSON(txFee)
 
-    // Convert additional transaction data.
-    const transactionDataCase = transaction.getTransactionDataCase()
-    switch (transactionDataCase) {
-      case Transaction.TransactionDataCase.PAYMENT: {
-        const payment = transaction.getPayment()
-        if (payment === undefined) {
-          return undefined
-        }
-        Object.assign(object, this.paymentToJSON(payment))
-        break
-      }
-      default:
-        throw new Error('Unexpected transactionDataCase')
+    // Set sequence numbers
+    object.Sequence = transaction.getSequence()?.getValue() ?? 0
+    object.LastLedgerSequence =
+      transaction.getLastLedgerSequence()?.getValue() ?? 0
+
+    const signingPubKeyBytes = transaction
+      .getSigningPublicKey()
+      ?.getValue_asU8()
+    if (signingPubKeyBytes) {
+      object.SigningPubKey = Utils.toHex(signingPubKeyBytes)
+    }
+
+    const additionalTransactionData = getAdditionalTransactionData(transaction)
```

Comment:
made a new `getAdditionalTransactionData()` helper to make things easier to reason about.